thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/rootcause/impl/DimensionPipeline.java

```diff
@@ -0,0 +1,146 @@
+import com.linkedin.thirdeye.dataframe.DoubleSeries;
+import com.linkedin.thirdeye.dataframe.Series;
+import com.linkedin.thirdeye.dataframe.StringSeries;
+import com.linkedin.thirdeye.datalayer.bao.DatasetConfigManager;
+import com.linkedin.thirdeye.datalayer.bao.MetricConfigManager;
+import com.linkedin.thirdeye.datalayer.dto.DatasetConfigDTO;
+import com.linkedin.thirdeye.datalayer.dto.MetricConfigDTO;
+import com.linkedin.thirdeye.rootcause.ExecutionContext;
+import com.linkedin.thirdeye.rootcause.Pipeline;
+import com.linkedin.thirdeye.rootcause.PipelineResult;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * Sample implementation of a pipeline for identifying relevant dimensions by performing
+ * contribution analysis. The pipeline first fetches the Current and Baseline entities and
+ * MetricEntities in the search context. It then maps the metrics to ThirdEye's internal database
+ * and performs contribution analysis using a {@code DimensionScorer).
+ *
+ * @see DimensionScorer
+ */
+public class DimensionPipeline implements Pipeline {
+  private static final Logger LOG = LoggerFactory.getLogger(DimensionPipeline.class);
+
+  public static final int DEFAULT_CUTOFF = 20;
+
+  static final String KEY = "key";
+  static final String DIMENSION = DimensionScorer.DIMENSION;
+  static final String VALUE = DimensionScorer.VALUE;
+  static final String COST = DimensionScorer.COST;
+
+  final MetricConfigManager metricDAO;
+  final DatasetConfigManager datasetDAO;
+  final DimensionScorer scorer;
+
+  final int cutoffSize;
+
+  public DimensionPipeline(MetricConfigManager metricDAO, DatasetConfigManager datasetDAO,
+      DimensionScorer scorer) {
+    this.metricDAO = metricDAO;
+    this.datasetDAO = datasetDAO;
+    this.scorer = scorer;
+    this.cutoffSize = DEFAULT_CUTOFF;
+  }
+
+  public DimensionPipeline(MetricConfigManager metricDAO, DatasetConfigManager datasetDAO,
+      DimensionScorer scorer, int cutoffSize) {
+    this.metricDAO = metricDAO;
+    this.datasetDAO = datasetDAO;
+    this.scorer = scorer;
+    this.cutoffSize = cutoffSize;
+  }
+
+  @Override
+  public String getName() {
+    return this.getClass().getSimpleName();
+  }
+
+  @Override
+  public PipelineResult run(ExecutionContext context) {
+    Set<MetricEntity> metricsEntities = EntityUtils.filterContext(context, MetricEntity.class);
+
+    TimeRangeEntity current = TimeRangeEntity.getContextCurrent(context);
+    TimeRangeEntity baseline = TimeRangeEntity.getContextBaseline(context);
+
+    DataFrame dfScore = new DataFrame();
+    dfScore.addSeries(DIMENSION, StringSeries.empty());
+    dfScore.addSeries(VALUE, StringSeries.empty());
+    dfScore.addSeries(COST, DoubleSeries.empty());
+
+    for(MetricEntity e : metricsEntities) {
+      MetricConfigDTO mdto = metricDAO.findByMetricAndDataset(e.getMetric(), e.getDataset());
```

Comment:
Minor comment, please try to keep variable names longer. It's quite difficult to follow the code with single character variable names like e